### PR TITLE
Add HPos and VPos::Default to allow the user to override axes label anchors

### PR DESCRIPTION
**Pull request by VincentFoulon80**
Proposition for #463

Add `HPos::Default` and `VPos::Default` variants, that allows to detect intentional changes to anchors and so prevent using unexpected change of position (The documentation didn't tell us axes overrides positioning values).

I decided to create new variants to address the issue of detecting that the user has changed the anchors of one axis's labels. (because the default Pos are `HPos::Left` and `VPos::Top`)

One other possibility is to change how the position is decided in `draw_impl`, based on the `FontTransform`. This may solve the issue as is but still won't allow customization of the position, and additionally we'd want to change the documentation in order to tell the user that positions are not used for axis labels.

### CODE REVIEW

File: plotters-backend/src/text.rs [hunk 1]
```diff
@@ -62,6 +62,9 @@ pub mod text_anchor {
     /// The horizontal position of the anchor point relative to the text.
     #[derive(Clone, Copy)]
     pub enum HPos {
+        /// Default value (Left), except chart axes that might provide a different pos.
+        /// Use another variant to override the default positionning
+        Default,
```
Comment (AaronErhardt) [inline plotters-backend/src/text.rs:67]: I think it makes sense to derive `Default` for both HPos and VPos since they already have a `Default` variant.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Looks good in general.

File: plotters-backend/src/text.rs (hunk 1 above)
Comment (VincentFoulon80) [reply plotters-backend/src/text.rs:67]: I am not sure I understood what you meant. Did you mean something like this ?
```diff
diff --git a/plotters-backend/src/text.rs b/plotters-backend/src/text.rs
index bcf3a7a..4acd931 100644
--- a/plotters-backend/src/text.rs
+++ b/plotters-backend/src/text.rs
@@ -60,10 +60,11 @@ impl<'a> From<&'a str> for FontFamily<'a> {
 /// ```
 pub mod text_anchor {
     /// The horizontal position of the anchor point relative to the text.
-    #[derive(Clone, Copy)]
+    #[derive(Default, Clone, Copy)]
     pub enum HPos {
         /// Default value (Left), except chart axes that might provide a different pos.
         /// Use another variant to override the default positionning
+        #[default]
         Default,
         /// Anchor point is on the left side of the text
         Left,
@@ -74,10 +75,11 @@ pub mod text_anchor {
     }

     /// The vertical position of the anchor point relative to the text.
-    #[derive(Clone, Copy)]
+    #[derive(Default, Clone, Copy)]
     pub enum VPos {
         /// Default value (Top), except chart axes that might provide a different pos
         /// Use another variant to override the default positionning
+        #[default]
         Default,
         /// Anchor point is on the top of the text
         Top,
@@ -88,7 +90,7 @@ pub mod text_anchor {
     }

     /// The text anchor position.
-    #[derive(Clone, Copy)]
+    #[derive(Default, Clone, Copy)]
     pub struct Pos {
         /// The horizontal position of the anchor point
         pub h_pos: HPos,
@@ -111,22 +113,6 @@ pub mod text_anchor {
         pub fn new(h_pos: HPos, v_pos: VPos) -> Self {
             Pos { h_pos, v_pos }
         }
-
-        /// Create a default text anchor position (top left).
-        ///
-        /// - **returns** The default text anchor position
-        ///
-        /// ```rust
-        /// use plotters_backend::text_anchor::{Pos, HPos, VPos};
-        ///
-        /// let pos = Pos::default();
-        /// ```
-        pub fn default() -> Self {
-            Pos {
-                h_pos: HPos::Default,
-                v_pos: VPos::Default,
-            }
-        }
     }
 }
```

**Review by AaronErhardt — APPROVED** [review]
Looks good, thanks!

**Comment by cszach** [comment]
Was this change included in 0.3.5? I used `HPos::Right, VPos::Center` like the code in #463 but still got the result like in the example plot.

**Comment by AaronErhardt** [comment]
No, this was not included because is was merged into `next-release-devel` instead of master. Historically, master is considered the stable branch and `next-release-devel` is for the next breaking semver release. I'm sorry if this caused confusion, I hope we can clean this up soon and switch to better branch names.

**Comment by cszach** [comment]
Thank you for the clarification! So at the moment how do I align text properly? Some text are getting into the graphing area and I want them to align nicely below the axis. Thanks in advance!

image

**Comment by AaronErhardt** [comment]
I think the alignment is still broken both on `master` and crates.io, so you need to use the `next-release-devel` branch. To make this work properly, this fix would need to be backported to the master branch and then we can publish a new release.

**Pull request merged**